### PR TITLE
Implement mnemonic phrase backup popup

### LINKED ISSUE #22: Implement the mnemonic phrase backup curtain page
We need to implement the full UX to ask users to backup their mnemonic phrase at a certain condition. It will be like this.

### User flow
1. At a certain condition, a curtain page will be displayed
2. The curtain page requires the user to complete the backup process before allowing him/her to continue using the app.
3. If the curtain page is passed, the user will be able to continue using the app. Optionally getting some rewards as finishing a quest.

### Conditions on triggering the curtain page
1. The user has made a certain number of transactions, e.g. after 5 posts or reactions
2. The user account has a certain value of assets
3. The user is going to operate some asset related operations, e.g. staking, which will be directed to Telescope in the future.

### Rewards
1. Drop a certain number of DSM to the account
2. Get a Badge

@RiccardoM @bragaz Let's discuss. We will also need @Rooty2020 to help prepare the design.

### PR DESCRIPTION
**Pull request by ryuash**
## Description

closes #22 
## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit tests.
- [x] Run flutter format.
- [x] Updated the documentation.
- [x] Added an entry to the `CHANGELOG.md` file.
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 - Implemented users jazz icons instead of old identicons
 - Implemented the ability of exporting and importing the mnemonic as an encrypted object safely
 - Removed the ability to manually set the app theme. Now it is the same set system-wide.
+- Add initial backup phrase feature
```
Comment (rmontagnin) [inline CHANGELOG.md]: I would be more specific here, maybe something like

> Added a popup to remind users of backup up their mnemonic phrase

File: ios/Flutter/.last_build_id [hunk 2]
```diff
@@ -0,0 +1 @@
+e4fdd8b222e094196d232258e242016b
```
Comment (rmontagnin) [inline ios/Flutter/.last_build_id]: I think we can remove this file from git all togheter. Or is this needed somewhere? If not, just run

```
git rm --cached ios/Flutter/.last_build_id
```

File: ios/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -313,7 +313,6 @@
 /* Begin XCBuildConfiguration section */
```
Comment (rmontagnin) [inline ios/Runner.xcodeproj/project.pbxproj:313]: I think we can remove also this file from being tracked into git: 

```
git rm --cached ios/Runner.xcodeproj/project.pbxproj
```

File: lib/ui/blocs/navigator/navigator_bloc.dart [hunk 4]
```diff
@@ -77,7 +79,8 @@ class NavigatorBloc extends Bloc<NavigatorEvent, void> {
   }
 
   void _mapNavigateToHomeEventToState() {
-    _navigatorKey.currentState.pushAndRemoveUntil(
+    dynamic pushMethod = _navigatorKey.currentState.pushAndRemoveUntil;
+    pushMethod(
```
Comment (rmontagnin) [inline lib/ui/blocs/navigator/navigator_bloc.dart]: Why did you declare the function as a variable? I think it makes the code harder to comprehend and it's better to switch to as it was before

```suggestion
    _navigatorKey.currentState.pushAndRemoveUntil(
```

File: lib/ui/blocs/navigator/navigator_bloc.dart [hunk 5]
```diff
@@ -144,16 +147,21 @@ class NavigatorBloc extends Bloc<NavigatorEvent, void> {
...
-  void _handleNavigateToShowMnemonic() async {
+  void _handleNavigateToShowMnemonicAuth(
+      NavigateToShowMnemonicAuth event) async {
+    dynamic pushMethod = _navigatorKey.currentState.push;
+
     final method = await _getAuthenticationMethodUseCase.get();
+
     if (method is BiometricAuthentication) {
-      await _navigatorKey.currentState.push(MaterialPageRoute(
+      await pushMethod(MaterialPageRoute(
```
Comment (rmontagnin) [inline lib/ui/blocs/navigator/navigator_bloc.dart]: As above, I think it's better to switch back this to as it was previously, to make the code easier to read

File: lib/ui/blocs/navigator/navigator_event.dart [hunk 6]
```diff
@@ -100,3 +107,7 @@ class NavigateToUserDetails extends NavigatorEvent {
   @override
   List<Object> get props => [user];
 }
+
+class NavigateToShowMnemonicVisualiser extends NavigatorEvent {}
+
+class NavigateToConfirmMnemonicBackupPhrase extends NavigatorEvent {}
```
Comment (rmontagnin) [inline lib/ui/blocs/navigator/navigator_event.dart]: Can you add the docs for these classes? I can't understand how they are used and when they are emitted

File: lib/ui/blocs/navigator/navigator_event.dart [hunk 7]
```diff
@@ -71,7 +71,14 @@ class NavigateToPostDetails extends NavigatorEvent {
 class NavigateToWallet extends NavigatorEvent {}
 
 /// Tells the Bloc to navigate to the screen that allows to show the mnemonic.
-class NavigateToShowMnemonic extends NavigatorEvent {}
+class NavigateToShowMnemonicAuth extends NavigatorEvent {
+  final bool backupPhrase;
```
Comment (rmontagnin) [inline lib/ui/blocs/navigator/navigator_event.dart:76]: Can you add the documentation for this field? I don't get how it's used

File: lib/ui/blocs/recover_account/recover_account_event.dart [hunk 8]
```diff
@@ -51,3 +51,5 @@ class ChangeFocus extends RecoverAccountEvent {
   @override
   List<Object> get props => [focusedField, currentText];
 }
+
+class BackupMnemonicSuccess extends RecoverAccountEvent {}
```
Comment (rmontagnin) [inline lib/ui/blocs/recover_account/recover_account_event.dart]: Please add the documentation to this class

File: lib/ui/theme/theme.dart [hunk 9]
```diff
@@ -20,6 +21,7 @@ class PostsTheme {
           ),
         ),
         iconTheme: IconThemeData(color: colorScheme.primary),
+        brightness: brightnessTheme,
```
Comment (rmontagnin) [inline lib/ui/theme/theme.dart]: Why have you added this?

File: lib/usecases/account/usecase_set_mnemonic_backup_popup.dart [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+  SetMnemonicBackupPopupUseCase({
+    @required UserRepository userRepository,
+  })  : assert(userRepository != null),
+        _userRepository = userRepository;
+
+  Future<bool> check() {
+    return _userRepository.shouldDisplayMnemonicBackupPopup();
+  }
+}
```
Comment (rmontagnin) [inline lib/usecases/account/usecase_set_mnemonic_backup_popup.dart]: Can you please document this use case?

File: test/repositories/settings/settings_repository_impl_test.dart [hunk 11]
```diff
@@ -1,3 +1,4 @@
+import 'dart:async';
```
Comment (rmontagnin) [inline test/repositories/settings/settings_repository_impl_test.dart:1]: Is this necessary?

**Review by rmontagnin — CHANGES_REQUESTED** [review]
First review, just asking for some more documentation

File: lib/ui/blocs/navigator/navigator_bloc.dart (hunk 4 above)
Comment (ryuash) [reply lib/ui/blocs/navigator/navigator_bloc.dart]: Forgot to clean up. Originally i had something like this going on
```
dynamic pushMethod = a;
if (statement) {
  dynamic pushMethod = b;
}

pushMethod(...push);
```

File: lib/ui/blocs/navigator/navigator_bloc.dart (hunk 5 above)
Comment (ryuash) [reply lib/ui/blocs/navigator/navigator_bloc.dart]: same as above. Forgot to clean up

File: lib/ui/theme/theme.dart (hunk 9 above)
Comment (ryuash) [reply lib/ui/theme/theme.dart]: Fixes the white appbar color

106564211_559877284687760_420802576461646916_n

File: lib/repositories/settings/source_settings_local.dart [hunk 12]
```diff
@@ -6,5 +6,5 @@ abstract class LocalSettingsSource {
   Future<dynamic> get(String key);
 
   /// Returns a stream based on the given key
```
Comment (rmontagnin) [inline lib/repositories/settings/source_settings_local.dart]:
```suggestion
  /// Returns a stream that emits all future changes to the value of the setting having the given [key]
```

File: lib/usecases/settings/repository_settings.dart [hunk 13]
```diff
@@ -9,5 +9,5 @@ abstract class SettingsRepository {
   Future<dynamic> get(String key);
 
   /// [Stream] that emits a value each time the given key is saved in to shared preference.
```
Comment (rmontagnin) [inline lib/usecases/settings/repository_settings.dart]:
```suggestion
  /// Returns a [Stream] that emits a value each time the value for the setting having the given [key] is changed.
```

File: lib/usecases/settings/usecase_watch_setting.dart [hunk 14]
```diff
@@ -11,7 +11,7 @@ class WatchSettingUseCase {
         _settingsRepository = settingsRepository;
 
   /// Returns a stream that returns the value being saved by the given key in shared preference
```
Comment (rmontagnin) [inline lib/usecases/settings/usecase_watch_setting.dart]:
```suggestion
  /// Returns a [Stream] that emits all the values that will be stored to the setting having the given [key].
```

File: lib/sources/settings/local/local_settings_source.dart [hunk 15]
```diff
@@ -47,8 +46,8 @@ class LocalSettingsSourceImpl implements LocalSettingsSource {
 
   @override
   Stream<dynamic> watch(String key) {
-    return _controller.stream
-        .where((event) => key == event.key)
-        .map((event) => event.value);
+    return _controller.stream.where((event) {
+      return (key == event.key);
+    }).map((event) => event.value);
```
Comment (rmontagnin) [inline lib/sources/settings/local/local_settings_source.dart]: I think we can simplify this

```suggestion
    return _controller.stream
      .where((event) => key == event.key)
      .map((event) => event.value);
```

File: lib/ui/blocs/home/home_bloc.dart [hunk 16]
```diff
@@ -75,6 +75,7 @@ class HomeBloc extends Bloc<HomeEvent, HomeState> {
       _watchSettingUseCase
           .watch(key: SettingKeys.TX_AMOUNT)
           .listen((value) async {
+        print('home bloc value: $value');
```
Comment (rmontagnin) [inline lib/ui/blocs/home/home_bloc.dart]: Can we remove this?

**Review by rmontagnin — COMMENTED** [review]
One comment to remove. Also can you add an entry inside the `distribution/whatsnew/whatsnew-en-US` file? Something maybe like 

```
• Added mnemonic phrase backup popup reminder
```

File: lib/ui/localization/export.dart [hunk 17]
```diff
@@ -292,6 +292,24 @@ Currently there are no reactions to this post yet.
 Be the first to add one!
 """;
   String reactionsTabLabel(int reactions) => "Reactions $reactions";
+
+  // mnemonic backup phrase
+  String get mnemonicBackupBody1 => 'Hi,';
+  String get mnemonicBackupBody2 => """
+Glad you see you here,
```
Comment (rmontagnin) [inline lib/ui/localization/export.dart]:
```suggestion
Glad to see you here,
```

File: test/sources/settings/local/local_settings_source_test.dart [hunk 18]
```diff
@@ -67,8 +67,8 @@ void main() {
       ]),
     );
 
-    repository.save(SettingKeys.TX_AMOUNT, 1);
-    repository.save(SettingKeys.BACKUP_POPUP_PERMISSION, false);
+    unawaited(repository.save(SettingKeys.TX_AMOUNT, 1));
+    unawaited(repository.save(SettingKeys.BACKUP_POPUP_PERMISSION, false));
```
Comment (rmontagnin) [inline test/sources/settings/local/local_settings_source_test.dart:72]: Why do we have these? Is it better to replace `unwaited` with `await` maybe?

**Review by rmontagnin — APPROVED** [review]
LGTM

**Pull request merged**